src/System.Windows.Forms/src/System/Windows/Forms/ComboBox.ComboBoxUiaTextProvider.cs

```diff
@@ -0,0 +1,421 @@
+                    return pt;
+                }
+
+                // get the UL corner of the character
+                pt = GetPositionFromCharIndex(startCharIndex);
+
+                // add the width of the character at that position.
+                if (GetTextExtentPoint32(ch, out Size size))
+                {
+                    pt.X += size.Width;
+                }
+
+                return pt;
+            }
+
+            public override void GetVisibleRangePoints(out int visibleStart, out int visibleEnd)
+            {
+                visibleStart = 0;
+                visibleEnd = 0;
+
+                if (!_owningComboBox.IsHandleCreated || IsDegenerate(_owningComboBox.ClientRectangle))
+                {
+                    return;
+                }
+
+                Rectangle rectangle = GetFormattingRectangle();
+                if (IsDegenerate(rectangle))
+                {
+                    return;
+                }
+
+                // Formatting rectangle is the boundary, which we need to inflate by 1
+                // in order to read characters within the rectangle
+                Point ptStart = new Point(rectangle.X + 1, rectangle.Y + 1);
+                Point ptEnd = new Point(rectangle.Right - 1, rectangle.Bottom - 1);
+
+                visibleStart = GetCharIndexFromPosition(ptStart);
+                visibleEnd = GetCharIndexFromPosition(ptEnd) + 1; // Add 1 to get a caret position after received character
+
+                return;
+
+                bool IsDegenerate(Rectangle rect)
+                    => rect.IsEmpty || rect.Width <= 0 || rect.Height <= 0;
+            }
+
+            public override bool LineScroll(int charactersHorizontal, int linesVertical)
+                // If the EM_LINESCROLL message is sent to a single-line edit control, the return value is FALSE.
+                => false;
+
+            public override void SetSelection(int start, int end)
+            {
+                if (!_owningComboBox.IsHandleCreated)
+                {
+                    return;
+                }
+
+                if (start < 0 || start > TextLength)
+                {
+                    Debug.Fail("SetSelection start is out of text range.");
+                    return;
+                }
+
+                if (end < 0 || end > TextLength)
+                {
+                    Debug.Fail("SetSelection end is out of text range.");
+                    return;
+                }
+
+                SendMessageW(_ownerChildEdit, (WM)EM.SETSEL, (IntPtr)start, (IntPtr)end);
+            }
+
+            private RECT GetFormattingRectangle()
+            {
+                Debug.Assert(_owningComboBox.IsHandleCreated);
+
+                // Send an EM_GETRECT message to find out the bounding rectangle.
+                RECT rectangle = new RECT();
+                SendMessageW(_ownerChildEdit, (WM)EM.GETRECT, IntPtr.Zero, ref rectangle);
+                return rectangle;
```

Comment:
nit: add a blank line
```suggestion
                SendMessageW(_ownerChildEdit, (WM)EM.GETRECT, IntPtr.Zero, ref rectangle);
                
                return rectangle;
```